docs/testing-framework.md

```diff
@@ -0,0 +1,438 @@
+---
+title: Cadence Testing Framework
+---
+
+Cadence testing framework provides a convenient way to write tests for Cadence programs in Cadence.
+This functionality is provided by the Test standard-library.
+
+Note: Test standard-library can only be used off-chain. e.g: In CLI
+
+## Test Standard Library
+
+Testing standard library can be imported directly into the test script.
+```cadence
+import Test
+```
+
+## Assertion
+### assert
+```cadence
+fun assert(_ condition: Bool, _ message: String)
+```
+Fails a test-case if the given condition is false, and reports a message which explains how the condition is false.
+
+The message argument is optional.
+
+### fail
+```cadence
+fun fail(_ message: string)
+```
+Immediately fails a test-case, with a message explaining the reason to fail the test.
+
+The message argument is optional.
+
+### expect
+The `expect` function tests a value against a matcher (see [matchers](#matchers) section), and fails the test if it's not a match.
+
+```cadence
+fun expect(_ value: AnyStruct, _ matcher: Matcher)
+```
+
+## Matchers
+A matcher is an object that consists of a test function and associated utility functionality.
+```cadence
```

Comment:
Here and below: Add spaces between headings, text, and code fences.

Reply:
markdown automatically adds spaces, so I didn't add them manually. But no harm in adding also 👍

Reply:
Added in e5f3d1f03849cb7031da6e2499885f5bb42da307